compiler/rustc_hir_typeck/src/fn_ctxt/suggestions.rs

```diff
@@ -3180,3 +3171,19 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
     }
 }
+
+/// For suggestion span, recursively try to look for the ancestor span which shares the same syntax
+/// context as the initially provided `span`. This will recursively look into local macros until the
+/// span inside the most ancestor local macro is found. It will stop recursing as soon as the
+/// syntax context of a potential parent callsite changes, such as if the potential parent callsite
+/// is in a foreign macro. This helps to prevent leaking implementation details from upstream
+/// crates and stdlib crates that the user likely have no control over.
+fn find_local_most_ancestor_suggestable_span(initial_span: Span) -> Span {
```

Comment:
Why is this different than [`Span::find_ancestor_in_same_ctxt`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_span/span_encoding/struct.Span.html#method.find_ancestor_in_same_ctxt)?

Reply:
Also https://github.com/rust-lang/rust/pull/112043/files#r1556509496

Reply:
Oh, I see. We find the *last* ancestor that shares the ctxt, not the first. Huh.

Reply:
Yeah, it was to get better error reporting for when there are multiple local nested macros.

Reply:
This method should go in `rustc_span`, as a method on `Span`.

Reply:
Do you have a better name for this? `find_local_most_ancestor_suggestable_span` is what I came up with but it feels not very accurate

Reply:
How about `find_oldest_ancestor_in_same_ctxt`?

Reply:
Sounds better than what I had 😆